jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/ConfigurationAsCodeNeedDockerTest.java

```diff
@@ -71,4 +75,43 @@ public void should_support_mysql_configuration(JenkinsRule r) throws Exception {
         }
     }
 
+    @Test
+    public void should_support_postgres_configuration(JenkinsRule r) throws Exception {
+
+        try {
+            POSTGRE_DB.start();
+            ExtensionList<CredentialsProvider> extensionList = r.jenkins.getExtensionList(CredentialsProvider.class);
+            extensionList.add(extensionList.size(), new GlobalPipelineMavenConfigTest.FakeCredentialsProvider());
+            String jdbcUrl = POSTGRE_DB.getJdbcUrl();
+
+            String yamlContent = toStringFromYamlFile(this, "configuration-as-code_postgresql.yml");
+            yamlContent = StringUtils.replace(yamlContent, "theJdbcUrl", jdbcUrl);
+
+            Path tmpYml = Files.createTempFile("pipeline-maven-plugin-test", "yml");
+            Files.write(tmpYml, yamlContent.getBytes());
+
+            ConfigurationAsCode.get().configure(singletonList(tmpYml.toFile().getAbsolutePath()));
+
+            GlobalPipelineMavenConfig config = r.jenkins.getExtensionList(GlobalPipelineMavenConfig.class).get(0);
+
+            assertThat(config.getJdbcUrl()).isEqualTo(jdbcUrl);
+            assertThat(config.getDaoClass()).isEqualTo(PipelineMavenPluginPostgreSqlDao.class.getName());
+
+            // we can't really test the PipelineMavenPluginMySqlDao is used as it is plenty of layers
+            // which doesn't expose the real implementation
+            assertThat(config.getDao().getClass()).isNotEqualTo(PipelineMavenPluginMySqlDao.class);
```

Comment:
`PipelineMavenPluginPostgresDao.class` ?

I know, useless, but just for coherency

Reply:
fixed 117fd54cb119e971b8474c85ce15931e6787cb03